docs(sources): add integrations directory with 31 source pages (ETL-980) (#739)
glassflow/clickhouse-etl: docs/app/sources/pubsub/page.mdx

---
title: 'Google Pub/Sub'
description: 'Stream events from Google Cloud Pub/Sub into ClickHouse with GlassFlow'
---
import { Callout } from 'nextra/components'

# Google Pub/Sub

Pub/Sub is a fully managed messaging service by Google Cloud for real-time event ingestion and delivery. It enables scalable, asynchronous communication between services using publish–subscribe patterns.

## Using Google Pub/Sub with GlassFlow

GlassFlow can ingest data from Google Cloud Pub/Sub topics into ClickHouse with the same pipeline capabilities available to all GlassFlow sources — deduplication, joins, filtering, and stateless transformations.

<Callout type="info">

> **Available on request.** Google Pub/Sub connectors are supported natively via the GlassFlow Enterprise Edition. To get started, [join our GlassFlow Slack community](https://glassflowhub.slack.com/join/shared_invite/zt-349m7lenp-IFeKSGfQwpJfIiQ7oyFFKg) to chat directly with our team, or [reach out](https://www.glassflow.dev/integrations#contact) to us with your use case, expected volumes, and timeline.

Reviewer: ```suggestion
**Available on request.** Google Pub/Sub connectors are natively supported in GlassFlow Enterprise Edition. To get started, [join our GlassFlow Slack community](https://glassflowhub.slack.com/join/shared_invite/zt-349m7lenp-IFeKSGfQwpJfIiQ7oyFFKg) to chat directly with our team, or [reach out](https://www.glassflow.dev/integrations#contact) to us with your use case, expected volumes, and timeline.
```
